### ISSUE TITLE
Motors - So many options

### ISSUE THREAD

**Issue by Xaphus** [CenWIDev/sumobot-resources#7]
I'm lost with all choices for motors - rpm, torque, mounting options, dc vs stepped.  Maybe someone with some more knowledge in this area could do a 10 min presentation next time we meet?  I want to start assembling playing with a build, but the cost of parts can add up quickly.

**Comment by Erik-Bennett** [comment]
Great question. The robotics competitions I'm familiar with generally start with a base kit (FIRST, VeX) for new teams, and it's questions like this that highlight why.

Motor selection depends on a number of factors including strategy. In watching video recently, I've seen radically different designs win - exceptionally quick sumobots for fast, momentum-based attacks, in contrast with bulkier, slower, high-torque pushers. They're clearly using different motor / controller systems in the robots, so there is no one right answer.

There are numerous forums with discussions on sumobot motor selection, and some more-general guides for robot motor selection on Youtube:
- [Build a REAL Robot Episode 3 - Selecting Motors](https://www.youtube.com/watch?v=q-LSxYCuaso)
- [RoboBench: Choosing Motors & Servos](https://www.youtube.com/watch?v=SR1fkrsq2L0)

**Comment by daniel-packard** [comment]
I'm tempted to get a pile of these, which match the motors and drivers provided in the sparkfun inventors kits:

* https://www.sparkfun.com/products/14450
* https://www.sparkfun.com/products/13302

Pros: 
* circuit examples and code examples from the inventors kit will work without modification
* compatible with other hardware from sparkfun (e.g. this chassis : https://www.sparkfun.com/products/13301)

Cons: 
* probably not the best overall choice in terms of performance, etc

Can get 20 motors and 10 motor dirvers for about $100 

... thoughts?

**Comment by Erik-Bennett** [comment]
Just a thought - I wonder if some of the exceptionally-quick competition sumobots  (e.g., wedge-momentum designs) are using brushless RC motors and ESCs.  Using RC parts opens up a lot of options for motor and controllers, particularly if anyone was into RC hobby.  They can be [prohibitively expensive](http://blog.jsumo.com/maxon-motors-for-sumo-robots/), but there are budget versions.

This path appears to be well-trodden, so there's [lots of information](https://safe.duckduckgo.com/?q=arduino+to+rc+esc&kp=2&ia=web) available on interfacing Arduino with ESCs.

Aside - [discussion on the differences between brushed and brushless motors, specific to sumobots](https://www.reddit.com/r/engineering/comments/adtmg8/choosing_the_type_of_electric_motor_for_a_sumo/).

**Comment by Erik-Bennett** [comment]
Regarding the Sparkfun motors, I'm curious why the rotor shaft protrudes from the housing as much as it does. That, and the (penurious) teacher in me cringes a bit at the pricing of both, though given the time constraint, paying more is understandable.

**Comment by daniel-packard** [comment]
Yeah, that's my main thought -- what's realistic given the timeline and existing skillsets?

I'm pretty sure we can get everyone spun up with these motors in no time -- everywhere else I look, I see a pretty substantial learning curve (for myself at least)

That's not to say we'd prohibit people from branching out

**Comment by daniel-packard** [comment]
PS: if we can come up with a budget for Greg, I'm sure Create would pitch in (if not buy outright) the supplies. We just need to make up our minds on what we need.

I think that enough parts for ~10 robots for less than $300 total is a pretty reasonable goal ($30 a piece)... we should bounce off greg next time we get a chance

Also, we can check 3rd party vendors for cheaper (compatible) alternatives

**Comment by Xaphus** [comment]
> Regarding the Sparkfun motors, I'm curious why the rotor shaft protrudes from the housing as much as it does. That, and the (penurious) teacher in me cringes a bit at the pricing of both, though given the time constraint, paying more is understandable.

I believe the shaft is longer so you can add a Wheel Encoder Kit to it:
https://www.sparkfun.com/products/12629

**Comment by Erik-Bennett** [comment]
Oh, fancy - hall effect encoders, not the quadrature type. Presumably, they'd have significantly higher resolution being on the motor shaft (before the gears) too.  I've seen these on automobiles (wheel speed sensors) and in brushless motors, but LEGO and Vex (and others I've seen for the TTs) employ the optical interrupt style.
